packages/neon/neon/lib/src/theme/theme.dart

```diff
@@ -34,17 +34,26 @@ class AppTheme {
   final NeonTheme neonTheme;
 
   ColorScheme _buildColorScheme(final Brightness brightness) {
-    final primary = nextcloudTheme?.color != null ? HexColor(nextcloudTheme!.color) : neonTheme.colorScheme.primary;
-    final keepOriginalAccentColorOverride = keepOriginalAccentColor ? primary : null;
+    final primaryColor =
+        nextcloudTheme?.color != null ? HexColor(nextcloudTheme!.color) : neonTheme.colorScheme.primary;
+    final primaryColorOverride = useNextcloudTheme ? primaryColor : null;
+
+    final onPrimaryColor = nextcloudTheme?.colorText != null ? HexColor(nextcloudTheme!.colorText) : null;
+    final onPrimaryColorOverride = useNextcloudTheme ? onPrimaryColor : null;
+
     final oledBackgroundOverride = oledAsDark && brightness == Brightness.dark ? NcColors.oledBackground : null;
 
     return ColorScheme.fromSeed(
-      seedColor: primary,
+      seedColor: primaryColor,
       brightness: brightness,
     ).copyWith(
       background: oledBackgroundOverride,
-      primary: keepOriginalAccentColorOverride,
-      secondary: keepOriginalAccentColorOverride,
+      primary: primaryColorOverride,
+      onPrimary: onPrimaryColorOverride,
+      secondary: primaryColorOverride,
+      onSecondary: onPrimaryColorOverride,
+      tertiary: primaryColorOverride,
+      onTertiary: onPrimaryColorOverride,
     );
   }
```

Comment:
I think something similar to the follwoing is way more readable.
Having this set up we can also drop the useNextcloudTheme null check with useNextcloudTheme in the constructor.

```dart
  ColorScheme _buildColorScheme(final Brightness brightness) {
    final nextcloudTheme = this.nextcloudTheme;

    var colorScheme = ColorScheme.fromSeed(
      seedColor: neonTheme.colorScheme.primary,
      brightness: brightness,
    );

    if (nextcloudTheme != null) {
      final primaryColor = HexColor(nextcloudTheme.color);
      colorScheme = ColorScheme.fromSeed(
        seedColor: primaryColor,
        brightness: brightness,
      );

      if (useNextcloudTheme) {
        final onPrimaryColor = HexColor(nextcloudTheme.colorText);

        colorScheme = colorScheme.copyWith(
          primary: primaryColor,
          onPrimary: onPrimaryColor,
          secondary: primaryColor,
          onSecondary: onPrimaryColor,
          tertiary: primaryColor,
          onTertiary: onPrimaryColor,
        );
      }
    }

    if (oledAsDark && brightness == Brightness.dark) {
      colorScheme = colorScheme.copyWith(
        background: NcColors.oledBackground,
      );
    }

    return colorScheme;
  }
```